underfs/oss/src/main/java/tachyon/underfs/oss/OSSUnderFileSystem.java

```diff
@@ -58,129 +92,489 @@ public UnderFSType getUnderFSType() {
+    try {
+      if (isFolder(key)) {
+        String keyAsFolder = convertToFolderName(stripPrefixIfPresent(key));
+        mOssClient.deleteObject(mBucketName, keyAsFolder);
+      } else {
+        mOssClient.deleteObject(mBucketName, stripPrefixIfPresent(key));
+      }
+    } catch (ServiceException se) {
+      LOG.error("Failed to delete {}", key, se);
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * Gets the {@link StorageObject} representing the metadata of a key. If the key does not exist as
+   * a file or folder, null is returned
+   * @param key the key to get the object details of
+   * @return {@link StorageObject} of the key, or null if the key does not exist as a file or folder
+   */
+  private ObjectMetadata getObjectDetails(String key) {
+    try {
+      if (isFolder(key)) {
+        String keyAsFolder = convertToFolderName(stripPrefixIfPresent(key));
+        return mOssClient.getObjectMetadata(mBucketName, keyAsFolder);
+      } else {
+        return mOssClient.getObjectMetadata(mBucketName, stripPrefixIfPresent(key));
+      }
+    } catch (ServiceException e) {
+      return null;
+    }
+
+  }
+
+  /**
+   * Gets the parent key of the input key, or null if no parent exists
+   * @param key the key to get the parent of
+   * @return the the parent key
+   */
+  private String getParentKey(String key) {
+    // Root does not have a parent.
+    if (isRoot(key)) {
+      return null;
+    }
+    int separatorIndex = key.lastIndexOf(PATH_SEPARATOR);
+    if (separatorIndex < 0) {
+      return null;
+    }
+    return key.substring(0, separatorIndex);
+  }
+
+  /**
+   * Set the OSS Client Configuration
+   * @param tachyonConf The TachyonConf
+   * @return The {@link ClientConfiguration} hold the OSS Client config
+   */
+  private ClientConfiguration initializeOSSClientConfig(TachyonConf tachyonConf) {
+    ClientConfiguration ossClientConf = new ClientConfiguration();
+    ossClientConf.setConnectionTimeout(
+        tachyonConf.getInt("tachyon.underfs.oss.connection.timeout.ms"));
+    ossClientConf.setSocketTimeout(
+        tachyonConf.getInt("tachyon.underfs.oss.socket.timeout.ms"));
+    ossClientConf.setConnectionTTL(tachyonConf.getLong("tachyon.underfs.oss.connection.ttl"));
+    ossClientConf.setMaxConnections(tachyonConf.getInt("tachyon.underfs.oss.connection.max"));
+    return ossClientConf;
+  }
+
+  /**
+   * Determines if the key represents a folder. If false is returned, it is not guaranteed that the
+   * path exists.
+   * @param key the key to check
+   * @return true if the key exists and is a directory
+   */
+  private boolean isFolder(String key) {
+    key = key.endsWith(PATH_SEPARATOR) ? key.substring(0, key.length() - 1) : key;
+    // Root is a folder
+    if (isRoot(key)) {
+      return true;
+    }
+    try {
+      String keyAsFolder = convertToFolderName(stripPrefixIfPresent(key));
+      mOssClient.getObjectMetadata(mBucketName, keyAsFolder);
```

Comment:
yes, If the key doesn't exist, the getObjectMetadata call will throw an exception